js/js_test.go

```diff
@@ -597,3 +597,26 @@ func TestTypeSwitchJSObject(t *testing.T) {
 		}
 	}
 }
+
+type StructWithNonIdentifierJsTags struct {
+	object *js.Object
+	Name   string `js:"my name"`
```

Comment:
How about adding more test cases like starting with a char that is not for IDs?

Reply:
As there are such tests, the results should be changed?

Reply:
I've made the test more "exciting" - I don't think any more are required because if we were doing the wrong thing then the generated Javascript would not parse.